src/cmdstan/command_helper.hpp

```diff
@@ -186,11 +259,17 @@ std::vector<std::string> make_filenames(const std::string &filename,
                                         const std::string &type,
                                         unsigned int num_chains,
                                         unsigned int id) {
-  std::vector<std::string> names(num_chains);
-  auto base_sfx = get_basename_suffix(filename);
-  if (base_sfx.second.empty()) {
+  std::pair<std::string, std::string> base_sfx;
+  std::string fname = filename;
+  std::transform(fname.begin(), fname.end(), fname.begin(), ::tolower);
+  if (boost::algorithm::ends_with(fname, ".csv")
+      || boost::algorithm::ends_with(fname, ".json")) {
```

Comment:
I'm not sure I understand why these are hard coded?

This does seem to fix the example provided in #1213, at least such that output files are properly generated (the `.baz` portion vanishes into thin air, but I suppose this is fine?), there is definitely still some oddities around though. For example,

```
./bernoulli pathfinder save_single_paths=1 output file=foo.bar.baz. data file=bernoulli.data.json
``` 
(note the period at the end of the output file) generates the same files as 
```
./bernoulli pathfinder save_single_paths=1 output file=foo.bar.baz data file=bernoulli.data.json
```

I think some more tests of specifically this function would be helpful!

Reply:
more unit tests have been added.
more discussion below.

Reply:
I don't think that ".baz" will disappear into the air - see line 95 or `command_helper_test.cpp`

Reply:
After pulling, and running `./bernoulli pathfinder save_single_paths=1 output file=foo.bar.baz data file=bernoulli.data.json`:

```
$ ls
bernoulli            bernoulli.data.R  foo.bar.csv         foo.bar_path_1.json  foo.bar_path_2.json  foo.bar_path_3.json  foo.bar_path_4.json
bernoulli.data.json  bernoulli.stan    foo.bar_path_1.csv  foo.bar_path_2.csv   foo.bar_path_3.csv   foo.bar_path_4.csv
```

no `baz` to be found! This is concerning, because I agree the test suggests otherwise!

Reply:
This is caused by the `output_base` variable in `command.hpp`. Integration test may be needed, sadly